tools/build-tools/src/test/typeValidator/classValidator.spec.ts

```diff
@@ -0,0 +1,131 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { strict as assert } from "assert";
+import os from "os";
+import { Project, SourceFile } from "ts-morph"
+import {
+    BreakingIncrement,
+    checkMajorIncrement,
+    checkMinorIncrement,
+    DecompositionTypeData,
+    tryDecomposeTypeData,
+} from "./../../typeValidator/packageValidator";
+import { enableLogging } from "./../../typeValidator/validatorUtils"
+
+describe("Class", () => {
+    enableLogging(true);
+    let project: Project;
+    let pkgDir: string = os.tmpdir();
+    beforeEach(() => {
+        project = new Project({
+            skipFileDependencyResolution: true,
+        });
+        assert(project !== undefined);
+    });
+
+    function getTypeDataForSource(sourceFile: SourceFile): DecompositionTypeData {
+        let typeData: DecompositionTypeData;
+        for (const declarations of sourceFile.getExportedDeclarations().values()) {
+            typeData = { kind: "unknown", name: "typeName", node: declarations[0] } as any as DecompositionTypeData;
+            tryDecomposeTypeData(project.getTypeChecker(), typeData);
+            break;
+        }
+        return typeData!;
+    }
+
+    // scenario: added new default param (method signature changed but won't break existing code)
+    // expected result: minor breaking change
+    it("new default value added", () => {
+        const classOld =
+        `
+        export class asdf {
+            public qewr() { return "afg"; }
+        }
+        `;
+        const oldSourceFile = project.createSourceFile(`${pkgDir}/src/classOld.ts`, classOld);
+        const oldTypeData = getTypeDataForSource(oldSourceFile);
+        // cont oldClassData = decomposeClassDeclaration(project.getTypeChecker(), oldSourceFile);
+
+        const classNew =
+        `
+        export class asdf {
+            public qewr(param1 : string = "afg") { return "afg"; }
+        }
+        `;
+        const newSourceFile = project.createSourceFile(`${pkgDir}/src/classNew.ts`, classNew);
+        const newTypeData = getTypeDataForSource(newSourceFile);
+
+        let increment = checkMajorIncrement(project, pkgDir, oldTypeData, newTypeData);
+        assert(increment === BreakingIncrement.none);
+
+        increment = checkMinorIncrement(project, pkgDir, oldTypeData, newTypeData);
+        assert(increment == BreakingIncrement.minor);
```

Comment:
could you integrate the changes from commit https://github.com/heliocliu/FluidFramework-1/commit/f1dc41848e49b6de406a4a39da61cbca15a33173 which refactors this?  it got added after you forked from that branch